server/middleware/validation.js

```diff
@@ -139,24 +159,39 @@ function validationMiddleware(req, res, next) {
     }
     next();
 
-    function isFieldVisible(visibilityConditions, data) {
+    function isFieldVisible(visibilityConditions, hideConditions, data) {
+        let i;
         let isVisible = true;
         if (visibilityConditions) {
             isVisible = false;
-            for (var i = 0; i < visibilityConditions.length; i++) {
+            for (i = 0; i < visibilityConditions.length; i++) {
```

Comment:
I'd put `let i` here (I know if it were `var` JavaScript hoisting would mean it made no difference, but I think here it'd mean the variable disappears when we exit the `for` loop, which would be safer). See `isComponentVisible()` further down this PR.